src/components/VideoDetailsView/VideoDetailsView.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+import {
+  youtubeApiGetSpecificVideo,
+  youtubeApiSearchRelated,
+} from '../../clients/youtube-api';
+import RelatedVideoFeed from '../RelatedVideoFeed';
+import {
+  RelatedVideoSection,
+  VideoDescription,
+  VideoPlayer,
+  VideoSection,
+  VideoTitle,
+  Wrapper,
+} from './VideoDetailsView.styled';
+
+export default function VideoDetailsView(props) {
+  const { videoId } = props;
+  const [videoTitle, setVideoTitle] = useState('');
+  const [videoDescription, setVideoDescription] = useState('');
+  const [relatedVideos, setRelatedVideos] = useState([]);
+  useEffect(() => {
+    youtubeApiGetSpecificVideo(videoId)
+      .then((res) => {
+        if (res.status === 403) {
+          throw new Error('Youtube API limit has exceeded');
+        } else {
+          return res.json();
+        }
+      })
+      .then((json) => {
+        setVideoTitle(json.items[0].snippet.title);
+        setVideoDescription(json.items[0].snippet.description);
+      })
+      .catch((error) => {
+        alert(error.message);
+      });
+    youtubeApiSearchRelated(videoId)
+      .then((res) => {
+        if (res.status === 403) {
+          throw new Error('Youtube API limit has exceeded');
+        } else {
+          return res.json();
+        }
+      })
```

Comment:
```js
.then((res) => {
  if (res.status === 403) {
    throw new Error('Youtube API limit has exceeded');
  } else {
    return res.json();
  }
})
```
We could create a custom hook, e.g `useRequest`, to reuse this logic more easily and remove duplicating code.